test/jdk/tools/launcher/I18NArgTest.java

```diff
@@ -125,5 +134,23 @@ static void testCharacters(String... args) {
                 "expected:" + expected + " obtained:" + hexValue;
             throw new RuntimeException(message);
         }
+        if (args.length == 3) {
+            // verify the value of the system property matches the expected value
+            String sysPropName = args[2];
+            String sysPropVal = System.getProperty(sysPropName);
+            if (sysPropVal == null) {
+                throw new RuntimeException("Missing system property " + sysPropName);
+            }
+            String sysPropHexVal = "";
+            for (int i = 0; i < sysPropVal.length(); i++) {
+                sysPropHexVal = sysPropHexVal.concat(Integer.toHexString(sysPropVal.charAt(i)));
+            }
```

Comment:
It's OK as it stands, but this loop could be replaced with a `HexFormat` one-liner.

Reply:
I'm sorry, I can't replace this code with a HexFormat one-line. Do you have an example code?

Reply:
Extract to a byte array with the appropriate charset and then HexFormat the bytearray.

```
    var SysPropHexVal = HexFormat.of().formatHex(sysPropVal.getBytes(StandardCharsets.UTF_16));
```

Reply:
Thanks, Roger. You beat me to it 🙂

Reply:
Thank you, I fixed it. In addition, since the return value of formatHex is four digits, I fixed the second argument passed to execTest to match that format.